CrosswordMe.py

```diff
@@ -1,95 +1,208 @@
+                if word_fits_in_range(current_word, puzzle_x_position, x_activator, puzzle_y_position, y_activator):
+                    for word_position in range(len(current_word)):
+                        if not word_matches_position(puzzle, current_word, word_position, puzzle_x_position, x_activator, puzzle_y_position, y_activator) and not puzzle_position_is_empty(puzzle, word_position, puzzle_x_position, x_activator, puzzle_y_position, y_activator):
+                            #the word doesnt fit in this position, breaks out of word loop
+                            start_x_position = -1
+                            start_y_position = -1
+                            word_direction = -1
+                            break
 
-        word_position = 0 #resets the word_position. this is needed beacuse word_position counts through the length of every word being written
-        for array_position in range(start_y_position, end_y_position): #Loops through position within the array
+                    else:
+                        #for is exceted without a break --> the word fits in this position
+                        start_x_position = puzzle_x_position
+                        start_y_position = puzzle_y_position
+                        word_direction = fit_direction
+                        break #breaks out of for direction loop, if a fitting position has been found
+            if start_x_position > 0: #breaks out of for x loop, if a fitting position has been found
+                break
+        if start_x_position > 0: #breaks out of for y loop, if a fitting position has been found
+            break
 
-            puzzle[array_position][column] = words[word_index][word_position] #Writes the word at random position of the array
-            word_position += 1 #Fake loop for the word position
+    return start_x_position, start_y_position, word_direction
 
-    return puzzle
 
-def write_diagonal(words, word_index, puzzle):
-    if word_index == 0:
+def find_best_fit(puzzle, current_word):
+    #Finds the best fit in the current puzzle for the word and returns the postion and direction
+    #The function checks the postion and directions in random order. The first Highscore wins.
+    #The funtion return a random position is no overlap fits can be found
+    fit = 0
+
+    for puzzle_y_position in sample(range(LINE_COUNT), k = LINE_COUNT): #loops through every position in the lines in random order
+        for puzzle_x_position in sample(range(COLUMN_COUNT), k = COLUMN_COUNT): #loops through every position in the columns in random order
+            for fit_direction in sample(range(3), k = 3):
+
+                x_activator, y_activator = set_direction_parameters(fit_direction)
+
+
+                temp_fit = 0
+                if  word_fits_in_range(current_word, puzzle_x_position, x_activator, puzzle_y_position, y_activator):
+
+                    for word_position in range(len(current_word)):
```

Comment:
this is now the fourth level of for loop. Usually you don't even want two, and certainly not 3. Maybe put part of this in another function or structure it differently? For the first two loops maybe look at the `itertools` module